app/build.gradle

```diff
@@ -0,0 +1,79 @@
+apply plugin: 'com.android.application'
+apply plugin: 'kotlin-android'
+apply plugin: 'kotlin-android-extensions'
+
+android {
+    compileSdkVersion 28
+    buildToolsVersion "29.0.3"
+    ndkVersion "21.1.6352462"
+
+    defaultConfig {
+        applicationId "com.mathewsachin.fategrandautomata"
+        minSdkVersion 24
+        targetSdkVersion 28
+        versionCode Integer.valueOf(System.env.FGA_VERSION_CODE ?: 1)
+        versionName (System.env.FGA_VERSION_NAME ?: "v0.1.0")
+
+        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
+
+        ndk {
+           abiFilters "armeabi-v7a", "arm64-v8a"
+        }
+    }
+
+    signingConfigs {
+        debug {
+            storeFile file("fgadebug.keystore")
+            storePassword "123abc"
+            keyAlias "fgadebug"
+            keyPassword "123abc"
+        }
+        release {
+            storeFile file("fgautomata.keystore")
+            storePassword System.env.KEYSTORE_PASS
+            keyAlias "fgautomata"
+            keyPassword System.env.KEYSTORE_PASS
+        }
+    }
+
+    buildTypes {
+        debug {
+            debuggable true
+            applicationIdSuffix '.debug'
+            signingConfig signingConfigs.debug
+        }
+        release {
+            minifyEnabled true
+            shrinkResources true
+            proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
+            signingConfig signingConfigs.release
+        }
+        fork {
+            initWith release
+            applicationIdSuffix '.fork'
+            signingConfig signingConfigs.debug
+        }
+    }
+}
+
+repositories {
+    maven { url "https://dl.bintray.com/blazei/maven" }
+}
+
+dependencies {
+    implementation fileTree(dir: 'libs', include: ['*.jar'])
+    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
+    implementation 'androidx.appcompat:appcompat:1.0.2'
+    implementation 'androidx.core:core-ktx:1.0.2'
+    implementation 'androidx.constraintlayout:constraintlayout:1.1.3'
+    implementation 'org.opencv:opencv-android:3.+'
```

Comment:
```suggestion
    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8:$kotlin_version"
    implementation 'androidx.appcompat:appcompat:1.1.0'
    implementation 'androidx.core:core-ktx:1.2.0'
    implementation 'androidx.constraintlayout:constraintlayout:1.1.3'
    implementation 'org.opencv:opencv-android:3.4.1'
```

Reply:
Notice, https://github.com/openpnp/opencv#api specifies that you have to do `OpenCV.loadShared()` before any native library can be used.

Reply:
Does this support Android though?
I knew about this library but the README doesn't mention Android support

Reply:
I decided to give this library a try anyway, but the app crashes on `OpenCV.loadShared()`.
I think it doesn't have Android support. The android specific OpenCV functions are also missing from the library.

Reply:
Hmm, but how will we ever get OpenCV updates? Do we really have to include the SDK in our build?

Reply:
Hmm, but how will we ever get OpenCV updates? Do we really have to include the SDK in our build?

Reply:
we'll have to maintain a separate repo with OpenCV. I'll fork the library I'm currently using and update it to latest version of OpenCV.

Reply:
This one's quite up-to-date: https://github.com/iamareebjamal/opencv-android